user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/web/rest/AccountResource.java

```diff
@@ -0,0 +1,272 @@
+
+    public AccountResource(UserRepository userRepository, UserService userService, UserCreationService userCreationService,
+            PasswordService passwordService, MailServiceProducer mailServiceProducer) {
+        this.userRepository = userRepository;
+        this.userService = userService;
+        this.userCreationService = userCreationService;
+        this.passwordService = passwordService;
+        this.mailServiceProducer = mailServiceProducer;
+    }
+
+    /**
+     * the registration is only enabled when the configuration artemis.user-management.registration.enabled is set to true.
+     * A non existing entry or false mean that the registration is not enabled
+     * @return whether the registration is enabled or not
+     */
+    private boolean isRegistrationDisabled() {
+        return registrationEnabled.isEmpty() || Boolean.FALSE.equals(registrationEnabled.get());
+    }
+
+    /**
+     * Returns true if saml2 app password is disabled, false otherwise.
+     *
+     * @return true if saml2 app password is disabled, false otherwise
+     */
+    private boolean isSAML2Disabled() {
+        return !(saml2EnablePassword.isPresent() && Boolean.TRUE.equals(saml2EnablePassword.get()));
+    }
+
+    /**
+     * {@code POST  /register} : register the user.
+     *
+     * @param managedUserVM the managed user View Model.
+     * @throws InvalidPasswordException {@code 400 (Bad Request)} if the password is incorrect.
+     * @throws EmailAlreadyUsedException {@code 400 (Bad Request)} if the email is already used.
+     * @throws LoginAlreadyUsedException {@code 400 (Bad Request)} if the login is already used.
+     */
+    @PostMapping("/register")
+    @ResponseStatus(HttpStatus.CREATED)
+    public void registerAccount(@Valid @RequestBody ManagedUserVM managedUserVM) {
+
+        if (isRegistrationDisabled()) {
+            throw new AccessForbiddenException("User Registration is disabled");
+        }
+        if (isPasswordLengthInvalid(managedUserVM.getPassword())) {
+            throw new InvalidPasswordException();
+        }
+
+        SecurityUtils.checkUsernameAndPasswordValidity(managedUserVM.getLogin(), managedUserVM.getPassword());
+
+        if (allowedEmailPattern.isPresent()) {
+            Matcher emailMatcher = allowedEmailPattern.get().matcher(managedUserVM.getEmail());
+            if (!emailMatcher.matches()) {
+                throw new BadRequestAlertException("The provided email is invalid and does not follow the specified pattern", "Account", "emailInvalid");
+            }
+        }
+
+        User user = userService.registerUser(managedUserVM, managedUserVM.getPassword());
+        mailServiceProducer.sendActivationMail(user);
+    }
+
+    /**
+     * {@code GET  /activate} : activate the registered user.
+     *
+     * @param key the activation key.
+     * @throws RuntimeException {@code 500 (Internal Server Error)} if the user couldn't be activated.
+     */
+    @GetMapping("/activate")
```

Comment:
why this is a GET call, should not be POST? since it does not retrieve a data, it just simply triggers a method in the server.

Reply:
The following functionality was not created by me, I only moved it from Artemis to the new microservice. 
I agree this can be changed to POST request but I do not want to change the client right now because the PR is big enough. If you agree I can apply this change in the future PR which removes the functionality from Artemis.

Reply:
oh, I see. nvm, it can be changed in the follow-up.